Content.Shared/Mobs/Systems/AddCompOnMobStateChangeSystem.cs

```diff
@@ -0,0 +1,37 @@
+using Content.Shared.Mobs.Components;
+
+namespace Content.Shared.Mobs.Systems
+{
+    /// <summary>
+    ///     Adds ItemComponent to entity when it dies. Remove when it revives.
+    /// </summary>
+    public sealed class AddCompOnMobStateChangeSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<AddCompOnMobStateChangeComponent, MobStateChangedEvent>(OnMobStateChanged);
+        }
+
+        private void OnMobStateChanged(EntityUid uid, AddCompOnMobStateChangeComponent component, MobStateChangedEvent args)
+        {
+            if(!TryComp<MobStateComponent>(uid, out var mobState))
+            return;
+
+            if (mobState.CurrentState == component.MobState)
+            {
+                foreach (var compType in component.Components)
+                {
+                    EntityManager.AddComponents(uid, component.Components);
+                }
```

Comment:
don't need the foreach loop

Reply:
oh, i did some changes after adding foreach and forgot to remove it